### PR TITLE
fix(http-client-csharp): hard coded response body content type

### PR DESCRIPTION
**Pull request by archerzz**
This is the intermediate fix for problematic media type properties.
- replace hard-coded value for `BodyMediaType`
- fix a bug in deducing media type from type, array should be `json`

No code is changed.

part of https://github.com/microsoft/typespec/issues/4208
work around https://github.com/microsoft/typespec/issues/4225

### CODE REVIEW

File: packages/http-client-csharp/emitter/src/lib/operation-converter.ts [hunk 1]
```diff
@@ -236,7 +236,7 @@ function fromSdkHttpOperationResponses(
     responses.push({
       StatusCodes: toStatusCodesArray(range),
       BodyType: r.type ? fromSdkType(r.type, sdkContext, typeCache) : undefined,
-      BodyMediaType: BodyMediaType.Json,
+      BodyMediaType: getBodyMediaType(r.type), // TOOD: https://github.com/microsoft/typespec/issues/4225
```
Comment (archerzz) [inline packages/http-client-csharp/emitter/src/lib/operation-converter.ts:239]: this is still not accurate, but better than hard-coded value

File: packages/http-client-csharp/emitter/src/type/body-media-type.ts [hunk 2]
```diff
@@ -11,18 +10,3 @@ export enum BodyMediaType {
   Text = "Text",
   Xml = "Xml",
 }
-
-export function typeToBodyMediaType(type: Type | undefined) {
```
Comment (archerzz) [inline packages/http-client-csharp/emitter/src/type/body-media-type.ts:15]: un-used codes, finally we should dump `BodyMediaType`

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/authentication/api-key/tspCodeModel.json [hunk 3]
```diff
@@ -81,7 +81,7 @@
        "StatusCodes": [
         204
        ],
-       "BodyMediaType": "Json",
+       "BodyMediaType": "None",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/authentication/api-key/tspCodeModel.json:84]: for http 204, apparently the media type should be none

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/encode/bytes/tspCodeModel.json [hunk 4]
```diff
@@ -1716,7 +1716,7 @@
         "CrossLanguageDefinitionId": "TypeSpec.bytes",
         "Decorators": []
        },
-       "BodyMediaType": "Json",
+       "BodyMediaType": "Binary",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/encode/bytes/tspCodeModel.json:1719]: for `bytes`, the media type should be `Binary`

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/payload/media-type/tspCodeModel.json [hunk 5]
```diff
@@ -140,7 +140,7 @@
        "StatusCodes": [
         200
        ],
-       "BodyMediaType": "Json",
+       "BodyMediaType": "None",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/payload/media-type/tspCodeModel.json:143]: http 200 but no return content, so the media type should be `None`.

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/payload/media-type/tspCodeModel.json [hunk 6]
```diff
@@ -207,7 +207,7 @@
         "CrossLanguageDefinitionId": "TypeSpec.string",
         "Decorators": []
        },
-       "BodyMediaType": "Json",
+       "BodyMediaType": "Text",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/payload/media-type/tspCodeModel.json:210]: return a pure string, so the media type is `Text`. Note that this happens to be correct. Deducing media type from return data type is not accurate.

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/array/tspCodeModel.json [hunk 7]
```diff
@@ -270,13 +270,13 @@
...
        ],
-       "BodyMediaType": "Json",
+       "BodyMediaType": "None",
        "Headers": [],
        "IsErrorResponse": false
       }
      ],
      "HttpMethod": "PUT",
-     "RequestBodyMediaType": "None",
+     "RequestBodyMediaType": "Json",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/array/tspCodeModel.json:279]: `array` input type should be of media type `Json`

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/scalar/tspCodeModel.json [hunk 8]
```diff
@@ -127,7 +127,7 @@
         "CrossLanguageDefinitionId": "TypeSpec.string",
         "Decorators": []
        },
-       "BodyMediaType": "Json",
+       "BodyMediaType": "Text",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/type/scalar/tspCodeModel.json:130]: This is the regression after we remove hard-coded `Json` value. From the `ContentTypes` below, I think it should be `Json`. But luckily the generated codes are not impacted.

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/payload/xml/tspCodeModel.json [hunk 9]
```diff
@@ -756,7 +756,7 @@
        "BodyType": {
         "$ref": "4"
        },
-       "BodyMediaType": "Json",
+       "BodyMediaType": "Xml",
```
Comment (archerzz) [inline packages/http-client-csharp/generator/TestProjects/CadlRanch/http/payload/xml/tspCodeModel.json:759]: for xml input, the body media type should be `Xml`

File: packages/http-client-csharp/generator/TestProjects/CadlRanch/http/authentication/api-key/tspCodeModel.json (hunk 3 above)
Comment (JoshLove-msft) [reply packages/http-client-csharp/generator/TestProjects/CadlRanch/http/authentication/api-key/tspCodeModel.json:84]: Should it be none, or omitted entirely?

File: packages/http-client-csharp/emitter/src/lib/operation-converter.ts [hunk 10]
```diff
@@ -271,19 +271,40 @@ function toStatusCodesArray(range: number | HttpStatusCodeRange): number[] {
...
+    case "string":
+      return BodyMediaType.Text;
+    case "bytes":
+      return BodyMediaType.Binary;
+    default:
+      return BodyMediaType.None;
+  }
+}
+
+function getBodyMediaType(type?: SdkType, contentTypes?: string[]): BodyMediaType {
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/emitter/src/lib/operation-converter.ts:291]: Can we add some tests?

File: packages/http-client-csharp/emitter/src/lib/operation-converter.ts [hunk 11]
```diff
@@ -236,7 +236,7 @@ function fromSdkHttpOperationResponses(
     responses.push({
       StatusCodes: toStatusCodesArray(range),
       BodyType: r.type ? fromSdkType(r.type, sdkContext, typeCache) : undefined,
-      BodyMediaType: BodyMediaType.Json,
+      BodyMediaType: getBodyMediaType(r.type, r.contentTypes), // TODO: https://github.com/microsoft/typespec/issues/4225
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/emitter/src/lib/operation-converter.ts:239]: I'm confused as to what is left todo here.

**Comment by JoshLove-msft** [comment]
Hi @archerzz, wondering what the status of this PR is. I've left some comments so if you can respond hopefully we can get this merged soon.

**Comment by archerzz** [comment]
> Hi @archerzz, wondering what the status of this PR is. I've left some comments so if you can respond hopefully we can get this merged soon.

@JoshLove-msft I remember this PR was blocked due to some issues found (see the issues created in the conversations above). Since it's not a blocker, let me close this first. We can restore it when we've resolved the prerequisites first.

**Comment by JoshLove-msft** [comment]
> > Hi @archerzz, wondering what the status of this PR is. I've left some comments so if you can respond hopefully we can get this merged soon.
> 
> @JoshLove-msft I remember this PR was blocked due to some issues found (see the issues created in the conversations above). Since it's not a blocker, let me close this first. We can restore it when we've resolved the prerequisites first.

Can you provide a link to the issue that is a prerequisite? I thought the issues in the PR summary were being fixed by this PR.

**Pull request closed without merging**